components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2432,43 +2436,76 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
      * @param password      password of the event receiver
      * @param updatedStatus status of the stat publishing state
      */
-    public void callStatupdateService(String receiverUrl, String user, String password, boolean updatedStatus) {
+    public void callStatUpdateService(String receiverUrl, String user, String password, boolean updatedStatus) {
 
-        try {
-            gatewyEnvironments = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
-                    getAPIManagerConfiguration().getApiGatewayEnvironments();
+        //all mandatory parameters should not be null in order to start the process
+        if (receiverUrl != null && user != null && password != null) {
+
+            if (log.isDebugEnabled()) {
+                log.debug("Updating Stats publishing status of Store/Publisher domain to : " + updatedStatus);
+            }
+
+            //get the cluster message agent to publisher-store domain
+            ClusteringAgent clusteringAgent = ServiceReferenceHolder.getContextService().getServerConfigContext().
+                    getAxisConfiguration().getClusteringAgent();
 
-            for (String gatewayEnvironmentName : gatewyEnvironments.keySet()) {
-                Environment currentGatewayEnvironment = gatewyEnvironments.get(gatewayEnvironmentName);
+            if (clusteringAgent != null) {
+                //changing stat publishing status at other nodes via a cluster message
+                try {
+                    clusteringAgent.sendMessage(new StatUpdateClusterMessage(updatedStatus), true);
+                } catch (ClusteringFault clusteringFault) {
+                    //error is only logged because initially gateway has modified the status
+                    log.error("Failed to send cluster message to Publisher/Store domain " +
```

Comment:
Fixed in : https://github.com/chamindias/carbon-apimgt/commit/2917bb535a4686418310b787af361524ff352252